DiscountDiscover/AppDelegate.m

```diff
@@ -87,6 +87,21 @@ -(void)scheduleNotification:(Deal *)deal {
     [center addNotificationRequest:request withCompletionHandler:nil];
 }
 
+-(void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
+    
+    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+    DetailsViewController *detailsViewController = [storyboard instantiateViewControllerWithIdentifier:@"DetailsViewController"];
+    
+    NSData *dealData = response.notification.request.content.userInfo[@"Deal"];
+    Deal *deal = [NSKeyedUnarchiver unarchivedObjectOfClass:Deal.class fromData:dealData error:nil];
+    detailsViewController.deal = deal;
+    
+    SceneDelegate *sd = (SceneDelegate *)[UIApplication sharedApplication].connectedScenes.allObjects[0].delegate;
+    UITabBarController *tabController = (UITabBarController *)sd.window.rootViewController;
+    UINavigationController *navController = (UINavigationController *)tabController;
```

Comment:
So, is `sd.window.rootViewController` a `UINavigationController` or a `UITabBarController`? You should check if `[sd.window.rootViewController isKindOfClass:[UINavigationController class]]` before casting, else you could get an object does not recognize selector exception on the next line which would crash the app.